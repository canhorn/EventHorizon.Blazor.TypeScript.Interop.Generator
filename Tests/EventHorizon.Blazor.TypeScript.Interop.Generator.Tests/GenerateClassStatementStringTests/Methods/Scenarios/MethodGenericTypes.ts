declare module Generic.Properties {
    export interface CameraInputsMap<T> { }
    export class Observable<T> { }

    export class ExampleClass<TCamera> {
        attached(): CameraInputsMap<TCamera>;
        initWebVR(): Observable<CameraInputsMap>;
        thisReturned(): this;
    }
}
