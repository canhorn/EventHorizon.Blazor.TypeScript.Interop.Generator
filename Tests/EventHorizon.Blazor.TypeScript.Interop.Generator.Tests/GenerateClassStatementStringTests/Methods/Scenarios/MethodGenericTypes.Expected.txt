/// Generated - Do Not Edit
namespace Generic.Properties
{
    using System;
    using System.Collections.Generic;
    using System.Text.Json.Serialization;
    using System.Threading.Tasks;
    using EventHorizon.Blazor.Interop;
    using EventHorizon.Blazor.Interop.Callbacks;
    using EventHorizon.Blazor.Interop.ResultCallbacks;
    using Microsoft.JSInterop;

    
    
    [JsonConverter(typeof(CachedEntityConverter<ExampleClass<CachedEntity>>))]
    public class ExampleClass<TCamera> : CachedEntityObject where TCamera : CachedEntity, new()
    {
        #region Static Accessors

        #endregion

        #region Static Properties

        #endregion

        #region Static Methods

        #endregion

        #region Accessors

        #endregion

        #region Properties

        #endregion
        
        #region Constructor
        public ExampleClass() : base() { }

        public ExampleClass(
            ICachedEntity entity
        ) : base(entity)
        {
            ___guid = entity.___guid;
        }


        #endregion

        #region Methods
        public CameraInputsMapCachedEntity<TCamera> attached()
        {
            return EventHorizonBlazorInterop.FuncClass<CameraInputsMapCachedEntity<TCamera>>(
                entity => new CameraInputsMapCachedEntity<TCamera>() { ___guid = entity.___guid },
                new object[]
                {
                    new string[] { this.___guid, "attached" }
                }
            );
        }

        public Observable<CameraInputsMapCachedEntity> initWebVR()
        {
            return EventHorizonBlazorInterop.FuncClass<Observable<CameraInputsMapCachedEntity>>(
                entity => new Observable<CameraInputsMapCachedEntity>() { ___guid = entity.___guid },
                new object[]
                {
                    new string[] { this.___guid, "initWebVR" }
                }
            );
        }

        public ExampleClass<TCamera> thisReturned()
        {
            return EventHorizonBlazorInterop.FuncClass<ExampleClass<TCamera>>(
                entity => new ExampleClass<TCamera>() { ___guid = entity.___guid },
                new object[]
                {
                    new string[] { this.___guid, "thisReturned" }
                }
            );
        }
        #endregion
    }
}